File: Excel/800-899/873/873 Challenge.R
library(tidyverse)
library(readxl)

path <- "Excel/800-899/873/873 Smaller Angle Between Clock Hands.xlsx"
input <- read_excel(path, range = "A1:A51")
test <- read_excel(path, range = "B1:B51")

angle <- function(h, m) {
  d <- abs((30 * h + .5 * m) - 6 * m)
  ifelse(d <= 180, d, 360 - d)
}

result = input %>%
  mutate(
    h = ifelse(hour(Time) >= 12, hour(Time) - 12, hour(Time)),
    m = minute(Time)
  ) %>%
  mutate(`Answer Expected` = angle(h, m))

all.equal(result$`Answer Expected`, test$`Answer Expected`)

File: Excel/800-899/873/873 Challenge.py
import pandas as pd

path = "Excel/800-899/873/873 Smaller Angle Between Clock Hands.xlsx"
input = pd.read_excel(path, usecols="A", nrows=51)
test = pd.read_excel(path, usecols="B", nrows=51)

def angle(h, m):
    d = abs((30*h + 0.5*m) - 6*m)
    return min(d, 360-d)

input['h'] = input['Time'].apply(lambda t: t.hour%12)
input['m'] = input['Time'].apply(lambda t: t.minute)
input['Answer Expected'] = input.apply(lambda r: angle(r['h'], r['m']), axis=1)

print(input['Answer Expected'].equals(test['Answer Expected']))